src/events/start.js

```diff
@@ -44,13 +44,24 @@ module.exports = function(socket) {
         }
 
         // Provide the callback to call when successful
-        Games.create(room, (newRoom, game) => {
+        Games.create(room, async (newRoom, game) => {
             let io = Server.getIO();
 
             io.to(room.id).emit('game', JSON.stringify(game));
             console.log(`user ${user.id} started game on room ${room.id}`);
 
             console.log(`game:\n` + JSON.stringify(game, null, 2));
+
+            let sockets = await io.in(newRoom.id).fetchSockets();
+            if (!sockets) {
+                console.error('Failed to obtain sockets from room');
+            }
+
+            // Install event listener on all sockets
+            for (s of sockets) {
+                // Start listening to the following events
+                require('./vote_persona')(s);
```

Comment:
The Flutter and the client are not playing together.
Whenever the Room owner starts, there is an exception that excludes the other type of client.
If Flutter is the owner, the Web clients cannot send their vote, and vice versa.
I have no idea what is problem.....

When the Web is the owner:
```
  socket.io-redis received request {"requestId":"2c0kAc","type":5,"opts":{"rooms":["room_69916"],"except":[]}} +1ms
  socket.io-redis received response {"requestId":"2c0kAc","sockets":[{"id":"wXZzkPyO6wvSeVM2AAAT","handshake":{"headers":{"user-agent":"Dart/2.14 (dart:io)","connection":"Upgrade","cache-control":"no-cache","accept-encoding":"gzip","sec-websocket-version":"13","host":"192.168.50.187:3000","sec-websocket-extensions":"permessage-deflate; client_max_window_bits","sec-websocket-key":"aNNefHX/VNyjueXLFbpF9g==","upgrade":"websocket"},"time":"Sun Oct 17 2021 02:14:52 GMT+0200 (Central European Summer Time)","address":"::ffff:192.168.50.31","xdomain":false,"secure":false,"issued":1634429692874,"url":"/socket.io/?EIO=4&transport=websocket","query":{"EIO":"4","transport":"websocket"},"auth":{}},"rooms":["room_69916"],"data":{}}]} +0ms
  socket.io-redis received request {"requestId":"2c0kAc","type":5,"opts":{"rooms":["room_69916"],"except":[]}} +28ms
  socket.io-redis ignoring unknown request +1ms
  socket.io-redis ignoring unknown request +9ms
  socket.io-redis received response {"requestId":"2c0kAc","sockets":[]} +22ms
(node:95965) UnhandledPromiseRejectionWarning: TypeError: socket.on is not a function
    at module.exports (/home/yuuma/git/have_you_heard_back/src/events/vote_persona.js:9:12)
    at /home/yuuma/git/have_you_heard_back/src/events/start.js:63:42
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at emitUnhandledRejectionWarning (internal/process/promises.js:168:15)
    at processPromiseRejections (internal/process/promises.js:247:11)
    at processTicksAndRejections (internal/process/task_queues.js:96:32)
(node:95965) TypeError: socket.on is not a function
    at module.exports (/home/yuuma/git/have_you_heard_back/src/events/vote_persona.js:9:12)
    at /home/yuuma/git/have_you_heard_back/src/events/start.js:63:42
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)

```

When Flutter is the owner:
```
  socket.io-redis received request {"requestId":"oWtDpP","type":5,"opts":{"rooms":["room_29003"],"except":[]}} +1ms
  socket.io-redis ignoring unknown request +3ms
  socket.io-redis ignoring unknown request +2ms
  socket.io-redis received response {"requestId":"oWtDpP","sockets":[{"id":"zU8po9CX_TYWQR-ZAAAR","handshake":{"headers":{"host":"localhost:3000","user-agent":"Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:92.0) Gecko/20100101 Firefox/92.0","accept":"*/*","accept-language":"en-US,en;q=0.5","accept-encoding":"gzip, deflate","connection":"keep-alive","referer":"http://localhost:3000/","sec-fetch-dest":"empty","sec-fetch-mode":"cors","sec-fetch-site":"same-origin"},"time":"Sun Oct 17 2021 02:12:17 GMT+0200 (Central European Summer Time)","address":"::ffff:127.0.0.1","xdomain":false,"secure":false,"issued":1634429537054,"url":"/socket.io/?EIO=4&transport=polling&t=NoBbugW","query":{"EIO":"4","transport":"polling","t":"NoBbugW"},"auth":{}},"rooms":["room_29003"],"data":{}},{"id":"DAD5XNjboqbgLgRgAAAP","handshake":{"headers":{"host":"localhost:3000","user-agent":"Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:92.0) Gecko/20100101 Firefox/92.0","accept":"*/*","accept-language":"en-US,en;q=0.5","accept-encoding":"gzip, deflate","connection":"keep-alive","referer":"http://localhost:3000/","sec-fetch-dest":"empty","sec-fetch-mode":"cors","sec-fetch-site":"same-origin"},"time":"Sun Oct 17 2021 02:12:16 GMT+0200 (Central European Summer Time)","address":"::ffff:127.0.0.1","xdomain":false,"secure":false,"issued":1634429536688,"url":"/socket.io/?EIO=4&transport=polling&t=NoBbua9","query":{"EIO":"4","transport":"polling","t":"NoBbua9"},"auth":{}},"rooms":["room_29003"],"data":{}}]} +8ms
  socket.io-redis received response {"requestId":"oWtDpP","sockets":[]} +1ms
(node:95965) UnhandledPromiseRejectionWarning: TypeError: socket.on is not a function
    at module.exports (/home/yuuma/git/have_you_heard_back/src/events/vote_persona.js:9:12)
    at /home/yuuma/git/have_you_heard_back/src/events/start.js:63:42
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at emitUnhandledRejectionWarning (internal/process/promises.js:168:15)
    at processPromiseRejections (internal/process/promises.js:247:11)
    at processTicksAndRejections (internal/process/task_queues.js:96:32)
(node:95965) TypeError: socket.on is not a function
    at module.exports (/home/yuuma/git/have_you_heard_back/src/events/vote_persona.js:9:12)
    at /home/yuuma/git/have_you_heard_back/src/events/start.js:63:42
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

Tested with this PR and https://github.com/yuumasato/have_you_heard/pull/37

Reply:
I think I know the issue: I'm trying to make the handler for the owner socket to install listeners on other sockets, but each socket can be handled by a different process. The right way to do this is to make the handler for each socket to install the listener for ``vote persona`` for itself.

Reply:
It works now, :)